app/tests/spec/lib/validate.js

```diff
@@ -16,16 +16,66 @@ define(function (require, exports, module) {
 
   describe('lib/validate', function () {
     describe('isEmailValid', function () {
+      it('returns true for valid email', function () {
+        assert.isTrue(Validate.isEmailValid('a@b.c'));
+      });
+
+      it('returns false for email without a local part', function () {
+        assert.isFalse(Validate.isEmailValid('@b.c'));
+      });
+
       it('returns false for email without a domain', function () {
-        assert.isFalse(Validate.isEmailValid('a'));
+        assert.isFalse(Validate.isEmailValid('a@'));
       });
 
-      it('returns true for email with a one part domain', function () {
-        assert.isTrue(Validate.isEmailValid('a@b'));
+      it('returns false for email with a single-part domain', function () {
+        assert.isFalse(Validate.isEmailValid('a@b'));
       });
 
-      it('returns true for valid email', function () {
-        assert.isTrue(Validate.isEmailValid('a@b.c'));
+      it('returns true for email with 64-character local part', function () {
+        assert.isTrue(Validate.isEmailValid(createRandomHexString(64) + '@b.c'));
+      });
+
+      it('returns false for email with 65-character local part', function () {
+        assert.isFalse(Validate.isEmailValid(createRandomHexString(65) + '@b.c'));
+      });
+
+      it('returns true for email with 256 characters', function () {
+        assert.isTrue(Validate.isEmailValid('a@' + createRandomHexString(252) + '.b'));
+      });
+
+      it('returns false for email with 257 characters', function () {
+        assert.isFalse(Validate.isEmailValid('a@' + createRandomHexString(253) + '.b'));
+      });
+
+      it('returns true for valid emails from auth server tests', function () {
+        [
+          'tim@tim-example.net',
+          'a+b+c@example.com',
+          '#!?-@t-e-s-t.c-o-m',
```

Comment:
We should probably add a 3 part domain to ensure they pass - `a@subdomain.domain.com`